code/script.js

```diff
@@ -1,325 +1,380 @@
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === "job") {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yes, the dog works as a ${value}! Keeping all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] === value
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, the dog doesn't work as a ${value}! Removing all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] !== value
+      );
+    }
+  } else if (category === "accessories") {
+    if (keep) {
+      alert(
+        `Yes, the dog is wearing ${value}! Keeping everyone wearing ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((dog) =>
+        dog[category].includes(value)
+      );
+    } else {
+      alert(
+        `No, the dog doesn't wear ${value}! Removing everyone wearing ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => !dog[category].includes(value)
+      );
+    }
+  } else if (category === "other") {
+    if (keep) {
+      alert(`Yes, the dog is a ${value}! Keeping all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter((dog) =>
+        dog[category].includes(value)
+      );
+    } else {
+      alert(`No, the dog is not a ${value}! Removing all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => !dog[category].includes(value)
+      );
     }
   }
+  questions--; //Removes 1 guess from the guess counter
+  questionCounter.innerHTML = questions; //Shows game counter to player
+  generateBoard();
+  dropdown.selectedIndex = null; //Clears the dropdown from previously chosen values
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  if (questions <= 0) {
+    //alerts the user that it's time to make their guess, lets them make a final guess
+    alert(`Time to make a guess!`);
+  } else if (questions <= -1) {
+    //alerts user that they've run out of questions
+    winOrLoseText.innerHTML = `You've run out of questions!`;
+    winOrLose.style.display = "flex";
+    board.style.display = "none";
+  }
+};
 
```

Comment:
This function is a neat idea to handle the number of allowed questions, alerts included. However, I see a bug where the counter keeps counting below 0 and the user can still pick questions and  eliminate characters beyond the allowed questions. 

What I could suggest is to possibly hide the questions section when it's 0, or a solution of your choice.